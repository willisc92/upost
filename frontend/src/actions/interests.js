import API from "../utils/API";

// SET_USER_INTERESTS
export const setUserInterests = (userInterests) => ({
    type: "SET_USER_INTERESTS",
    userInterests
});

// START_SET_USER_INTERESTS
export const startSetUserInterests = () => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            API.get(`user-interests/${localStorage.getItem("user_id")}`)
                .then((result) => {
                    dispatch(setUserInterests(result.data.interests));
                    resolve(true);
                })
                .catch((error) => {
                    console.log("error in getting user interests", error);
                    reject(error);
                });
        });
    };
};

// EDIT_USER_INTERESTS
export const editUserInterests = (userInterests) => ({
    type: "EDIT_USER_INTERESTS",
    userInterests
});

// START_EDIT_USER_INTERESTS
export const startEditUserInterests = (userInterests) => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            API.put(`user-interests/${localStorage.getItem("user_id")}/`, {
                interests: userInterests
            })
                .then((result) => {
                    dispatch(editUserInterests(userInterests));
                    resolve(true);
                })
                .catch((error) => {
                    console.log("error in updating user interests", error);
                    reject(error);
                });
        });
    };
};

export const getAllInterests = () => {
    return (dispatch) => {
        return new Promise((resolve, reject) => {
            API.get(`interests/`)
                .then((result) => {
                    dispatch(setUserInterests(result.data));
                    resolve(true);
                })
                .catch((error) => {
                    console.log("error in getting Interests", error);
                    reject(error);
                });
        });
    };
};
