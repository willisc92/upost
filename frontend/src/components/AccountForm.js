import React from "react";
import { connect } from "react-redux";
import { SingleDatePicker } from "react-dates";
import Select from "react-select";
import moment from "moment";
import API from "../utils/API";

class AccountForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            calendarFocused: false,
            date: moment(),
            birthDate: moment(),
            isLoaded: false,
            interests: [],
            error: undefined,
            selectedInterests: []
        };
    }

    onDateChange = (birthDate) => {
        if (birthDate) {
            this.setState(() => ({ birthDate }));
        }
    };

    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };

    onInterestChange = (selectedOption) => {
        this.setState({ selectedInterests: selectedOption });
    };

    componentDidMount() {
        API.get("interests/").then(
            (result) => {
                const interests = result.data.map((interest) => {
                    return { value: interest.interest_tag, label: interest.interest_tag };
                });
                this.setState({
                    isLoaded: true,
                    interests: interests
                });
            },

            (error) => {
                this.setState({
                    isLoaded: true,
                    error
                });
            }
        );
    }

    onSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit({
            formInput: e.target,
            interests: this.state.selectedInterests.map((interest) => {
                return interest.value;
            }),
            date: this.state.date,
            birthDate: this.state.birthDate
        });
    };

    render() {
        return (
            <form onSubmit={this.onSubmit}>
                <p>Username*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.username &&
                    this.props.auth.error.username.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="username" />
                <p>150 characters of fewer. Letters, digits and @/./+/-/) only.</p>
                <p>Email Address</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.email &&
                    this.props.auth.error.email.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="email" />
                <p>First Name*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.first_name &&
                    this.props.auth.error.first_name.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="firstName" />
                <p>Middle Name</p>
                <input type="text" name="middleName" />
                <p>Last Name*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.last_name &&
                    this.props.auth.error.last_name.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="lastName" />
                <p>Birth date</p>
                <SingleDatePicker
                    date={this.state.date}
                    onDateChange={this.onDateChange}
                    focused={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={(day) => false}
                />
                <p>Country*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.country &&
                    this.props.auth.error.country.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="country" />
                <p>State*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.state &&
                    this.props.auth.error.state.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="state" />
                <p>Steet Name*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.street_name &&
                    this.props.auth.error.street_name.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="streetName" />
                <p>Postal Code*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.postal_code &&
                    this.props.auth.error.postal_code.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="postalCode" />
                <p>City*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.city &&
                    this.props.auth.error.city.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="city" />
                <p>Sex</p>
                <select name="sex">
                    <option>Male</option>
                    <option>Female</option>
                </select>
                <p>Phone Number</p>
                <input type="text" name="phoneNumber" />
                <p>Interests*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.interests &&
                    this.props.auth.error.interests.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <Select
                    value={this.state.selectedInterests}
                    onChange={this.onInterestChange}
                    options={this.state.interests}
                    isMulti
                    name="interests"
                />
                <p>Password</p>
                <p>Your password contain at least {this.props.minPasswordLength} characters.</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.password &&
                    this.props.auth.error.password.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="password" />
                <p>Password Confirmation*</p>
                {!!this.props.auth.error &&
                    !!this.props.auth.error.password &&
                    this.props.auth.error.password.map((error) => {
                        return <p key={error}>{error}</p>;
                    })}
                <input type="text" name="passwordConfirmation" />
                <br />
                <button>Sign Up</button>
            </form>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        auth: state.auth
    };
};

export default connect(mapStateToProps)(AccountForm);
