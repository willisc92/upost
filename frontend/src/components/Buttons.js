import React from "react";
import Button from "@material-ui/core/Button";
import { makeStyles } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme) => ({
    whiteButton: {
        color: "white"
    },
    whiteRedButton: {
        color: "white",
        backgroundColor: theme.palette.primary.main,
        margin: 5
    }
}));

const White = (WrappedComponent) => {
    const HOC = (props) => {
        const classes = useStyles();
        return <WrappedComponent {...props} className={classes.whiteButton} />;
    };

    return HOC;
};

const WhiteRed = (WrappedComponent) => {
    const HOC = (props) => {
        const classes = useStyles();
        return <WrappedComponent {...props} className={classes.whiteRedButton} />;
    };

    return HOC;
};

export const WhiteButton = White(Button);
export const WhiteRedButton = WhiteRed(Button);
