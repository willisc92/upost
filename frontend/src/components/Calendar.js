import { Calendar, momentLocalizer } from "react-big-calendar";
import moment from "moment";
import React from "react";
const localizer = momentLocalizer(moment);

export const MyCalendar = (props) => (
    <div className="calendar_container">
        <Calendar
            localizer={localizer}
            events={props.events}
            onSelectEvent={props.onSelectEvent}
            startAccessor="start"
            endAccessor="end"
            views={["month", "day", "week"]}
        />
    </div>
);
