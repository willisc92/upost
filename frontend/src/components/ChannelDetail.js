import React from "react";
import API from "../utils/API";

class ChannelDetail extends React.Component {
    state = {
        detail: {},
        isLoaded: false,
        error: null
    };

    componentDidMount() {
        API.get("channels/").then(
            (result) => {
                this.setState({
                    isLoaded: true,
                    channels: result.data
                });
            },

            (error) => {
                this.setState({
                    isLoaded: true,
                    error
                });
            }
        );
    }

    render() {
        const { error, isLoaded, items } = this.state;
        if (error) {
            return <div>Error: {error.message}</div>;
        } else if (!isLoaded) {
            return <div>Loading...</div>;
        } else {
            return (
                <div>
                    {this.state.channels.map((channel) => (
                        <div key={channel.channel_id}>
                            <h1>{channel.channel_name}</h1>
                            <span>{channel.creation_date}</span>
                        </div>
                    ))}
                </div>
            );
        }
    }
}

export default ChannelList;
