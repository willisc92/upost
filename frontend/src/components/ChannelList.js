import React from 'react';

class ChannelList extends React.Component {
    state = {
        channels: []
    };

    async componentDidMount() {
        try {
          const res = await fetch('http://127.0.0.1:8000/api/channels/');
          const channels = await res.json();
          this.setState({
            channels
          });
        } catch (e) {
          console.log(e);
        }
    }
    
    render() {
      return (
        <div>
          {this.state.channels.map((channel) => (
            <div key={channel.channel_id}>
              <h1>{channel.channel_name}</h1>
              <span>{channel.creation_date}</span>
            </div>
          ))}
        </div>
      );
    }
}

export default ChannelList; 

    