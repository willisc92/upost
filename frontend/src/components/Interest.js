import React from "react";

export class Interest extends React.Component {
    onClick = () => {
        this.props.changeIsSelected(this.props.interest.interest_tag);
    };

    render() {
        return (
            <div className="polaroid" onClick={this.onClick}>
                <svg className="polaroid__selector">
                    <circle
                        className="polaroid__circle"
                        cx={0}
                        cy={0}
                        r={22}
                        fill={this.props.interest.isSelected ? "red" : "grey"}
                    />
                </svg>
                <img className="polaroid__image" src={this.props.interest.image} />
                <h3 className="polaroid__title">{this.props.interest.interest_tag}</h3>
                <p className="polaroid__description">{this.props.interest.description}</p>
            </div>
        );
    }
}

export default Interest;
