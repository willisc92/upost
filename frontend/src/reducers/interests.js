// Interests Reducers
const interestsDefaultState = [];

export default (state = interestsDefaultState, action) => {
  switch (action.type) {
    case "SET_USER_INTERESTS":
      return {
        userInterests: action.userInterests
      };
    case "EDIT_USER_INTERESTS":
      return {
        userInterests: action.userInterests
      };
    default:
      return state;
  }
};
