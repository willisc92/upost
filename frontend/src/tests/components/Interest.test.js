import { Interest } from "../../components/Interest";
import { shallow } from "enzyme";
import React from "react";
import interests from "../fixtures/interests";

test("Should render an interest component", () => {
    const interest = interests[0];
    const wrapper = shallow(<Interest interest={interest} />);
    expect(wrapper).toMatchSnapshot();
});

test("Should call changeIsSelected", () => {
    const interest = interests[0];
    const changeIsSelectedSpy = jest.fn();
    const wrapper = shallow(<Interest interest={interest} changeIsSelected={changeIsSelectedSpy} />);
    wrapper.find("div").simulate("click");
    expect(changeIsSelectedSpy).toHaveBeenCalled();
    expect(changeIsSelectedSpy).toHaveBeenCalledWith(interests[0].interest_tag);
});
