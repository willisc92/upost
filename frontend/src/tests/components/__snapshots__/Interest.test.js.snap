// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render an interest component 1`] = `
<div
  className="interest"
  onClick={[Function]}
>
  <svg
    className="interest-selector"
  >
    <circle
      className="interest-circle"
      cx={0}
      cy={0}
      fill="grey"
      r={22}
    />
  </svg>
  <img
    className="interest-image"
    src="http://localhost:8000/media/interests/fa-image.png"
  />
  <p
    className="interest-title"
  >
    Interest 1
  </p>
  <p
    className="interest-description"
  >
    description
  </p>
</div>
`;
